File: backend/test_projects_api.py
import requests
import json

# 测试登录获取token
login_url = "http://localhost:5000/api/auth/login"
login_data = {
    "username": "admin",
    "password": "admin123"
}

try:
    # 登录获取token
    response = requests.post(login_url, json=login_data)
    if response.status_code == 200:
        token = response.json()['access_token']
        print(f"登录成功，token: {token[:20]}...")
        
        # 测试项目API
        headers = {
            'Authorization': f'Bearer {token}',
            'Content-Type': 'application/json'
        }
        
        projects_url = "http://localhost:5000/api/projects"
        response = requests.get(projects_url, headers=headers)
        
        if response.status_code == 200:
            result = response.json()
            print(f"项目API调用成功，返回数据: {result}")
            if isinstance(result, list):
                print(f"返回项目数量: {len(result)}")
                for project in result:
                    print(f"项目ID: {project['id']}, 名称: {project['name']}, 状态: {project['status']}")
            else:
                print(f"返回数据类型: {type(result)}")
        else:
            print(f"项目API调用失败，状态码: {response.status_code}")
            print(f"错误信息: {response.text}")
    else:
        print(f"登录失败，状态码: {response.status_code}")
        print(f"错误信息: {response.text}")
        
except Exception as e:
    print(f"测试过程中出现错误: {str(e)}")

File: vue-frontend/test_projects_api.js
import axios from 'axios';

// 创建axios实例
const api = axios.create({
  baseURL: 'http://localhost:5000/api/v1',
  headers: {
    'Content-Type': 'application/json',
  },
  withCredentials: true
});

// 请求拦截器
api.interceptors.request.use(
  config => {
    console.log('Request URL:', config.url);
    console.log('Request Method:', config.method);
    return config;
  },
  error => {
    console.error('Request Error:', error);
    return Promise.reject(error);
  }
);

// 响应拦截器
api.interceptors.response.use(
  response => {
    console.log('Response Status:', response.status);
    console.log('Response Data:', response.data);
    return response;
  },
  error => {
    console.error('Response Error:', error.response ? error.response.status : 'No response');
    console.error('Error Data:', error.response ? error.response.data : error.message);
    return Promise.reject(error);
  }
);

// 测试获取项目列表
async function testProjectsApi() {
  try {
    console.log('Testing GET /projects endpoint...');
    const response = await api.get('/projects');
    console.log('\nProjects API Test Result: SUCCESS');
    console.log('Number of projects returned:', response.data.length);
    
    // 先登录获取token，然后再测试需要认证的接口
    console.log('\nTesting login first to get authentication token...');
    const loginResponse = await api.post('/auth/login', {
      email: 'admin@bugtracker.com',
      password: 'admin123'
    });
    
    // 设置认证token
    const token = loginResponse.data.access_token;
    api.defaults.headers.common['Authorization'] = `Bearer ${token}`;
    
    console.log('\nTesting GET /projects with authentication...');
    const authResponse = await api.get('/projects');
    console.log('Authenticated Projects API Test Result: SUCCESS');
    console.log('Number of projects returned with auth:', authResponse.data.length);
    
  } catch (error) {
    console.error('\nProjects API Test Result: FAILED');
    console.error('Error Details:', error.message);
  }
}

// 运行测试
testProjectsApi();